Meduna and near sectors, destroyed tank(s) available as vehicle, using it crashes the game

#### Bug description

Version Label: Stracciatella v0.23.0-git+03452b628
Build 04.12.02

---

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
| v0.23.0-git+03452b628 | JA2 linux version  | not sure | not sure |

---

JA2S Sci-Fi mode.

To trigger the "Crashes on using tank as vehicle" assing one of mercs in sector N6 to a vehicle, do not use the hummer from the list, choose the tank.

The logs:

	2026-08-03T07:48:02.949315929Z [INFO] stracciatella::logger: Logging to file "/tmp/ja2.log"
	2026-08-03T07:48:02.96903398Z [WARN] stracciatella::mods: Could not read mod manifest for `nightops-maps` at location `"/home/needle/.ja2/mods/nightops-maps/manifest.json"`: No such file or directory (os error 2)
	2026-08-03T07:48:03.035193384Z [INFO] stracciatella::vfs: VFS layer 1: SlfFs { source: DirFsFile { "tilesets.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035220722Z [INFO] stracciatella::vfs: VFS layer 2: SlfFs { source: DirFsFile { "speech.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035236521Z [INFO] stracciatella::vfs: VFS layer 3: SlfFs { source: DirFsFile { "sounds.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035251971Z [INFO] stracciatella::vfs: VFS layer 4: SlfFs { source: DirFsFile { "radarmaps.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035267432Z [INFO] stracciatella::vfs: VFS layer 5: SlfFs { source: DirFsFile { "npcdata.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035283265Z [INFO] stracciatella::vfs: VFS layer 6: SlfFs { source: DirFsFile { "npc_speech.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035298627Z [INFO] stracciatella::vfs: VFS layer 7: SlfFs { source: DirFsFile { "music.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035313933Z [INFO] stracciatella::vfs: VFS layer 8: SlfFs { source: DirFsFile { "mercedt.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035329265Z [INFO] stracciatella::vfs: VFS layer 9: SlfFs { source: DirFsFile { "maps.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035344403Z [INFO] stracciatella::vfs: VFS layer 10: SlfFs { source: DirFsFile { "loadscreens.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035359656Z [INFO] stracciatella::vfs: VFS layer 11: SlfFs { source: DirFsFile { "laptop.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035374852Z [INFO] stracciatella::vfs: VFS layer 12: SlfFs { source: DirFsFile { "interface.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035390096Z [INFO] stracciatella::vfs: VFS layer 13: SlfFs { source: DirFsFile { "fonts.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035405328Z [INFO] stracciatella::vfs: VFS layer 14: SlfFs { source: DirFsFile { "faces.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035420552Z [INFO] stracciatella::vfs: VFS layer 15: SlfFs { source: DirFsFile { "data.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.03543582Z [INFO] stracciatella::vfs: VFS layer 16: SlfFs { source: DirFsFile { "cursors.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035451034Z [INFO] stracciatella::vfs: VFS layer 17: SlfFs { source: DirFsFile { "binarydata.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035466353Z [INFO] stracciatella::vfs: VFS layer 18: SlfFs { source: DirFsFile { "bigitems.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035481468Z [INFO] stracciatella::vfs: VFS layer 19: SlfFs { source: DirFsFile { "battlesnds.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035496756Z [INFO] stracciatella::vfs: VFS layer 20: SlfFs { source: DirFsFile { "anims.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035512034Z [INFO] stracciatella::vfs: VFS layer 21: SlfFs { source: DirFsFile { "ambient.slf" in "/opt/ja2/data" } }
	2026-08-03T07:48:03.035527152Z [INFO] stracciatella::vfs: VFS layer 22: DirFs { "/opt/ja2/data" }
	2026-08-03T07:48:03.035543824Z [INFO] stracciatella::vfs: VFS layer 23: DirFs { "/opt/ja2-straciatella.git/externalized" }
	2026-08-03T07:48:03.03556055Z [INFO] stracciatella::vfs: VFS layer 24: DirFs { "/opt/ja2-straciatella.git/mods/from-russia-with-love/data" }
	2026-08-03T07:48:03.035577482Z [INFO] stracciatella::vfs: VFS layer 25: DirFs { "/opt/ja2-straciatella.git/mods/generous-rebels/data" }
	2026-08-03T07:48:03.035594117Z [INFO] stracciatella::vfs: VFS layer 26: DirFs { "/opt/ja2-straciatella.git/mods/imp-quiz-honest-answers/data" }
	2026-08-03T07:48:03.035611001Z [INFO] stracciatella::vfs: VFS layer 27: DirFs { "/home/needle/.ja2/mods/kaerar-gun-pack/Data" }
	2026-08-03T07:48:03.035640614Z [INFO] externalized/DefaultContentManager.cc: JA2 Home Dir:                  '/home/needle/.ja2'
	2026-08-03T07:48:03.035656027Z [INFO] externalized/DefaultContentManager.cc: Root game resources directory: '/opt/ja2'
	2026-08-03T07:48:03.035675124Z [INFO] externalized/DefaultContentManager.cc: Extra data directory:          '/opt/ja2-straciatella.git'
	2026-08-03T07:48:03.035688932Z [INFO] externalized/DefaultContentManager.cc: Saved games directory:         '/home/needle/.ja2/SavedGames'
	2026-08-03T07:48:03.035702483Z [INFO] externalized/DefaultContentManager.cc: Temporary directory:           '/tmp/ja2-stracciatella-kvyft1xs'
	2026-08-03T07:48:03.03571677Z [INFO] externalized/ModPackContentManager.cc: Enabled mods                    'from-russia-with-love, generous-rebels, imp-quiz-honest-answers, kaerar-gun-pack'
	2026-08-03T07:48:03.681454492Z [INFO] game/GameLoop.cc: Version Label: Stracciatella v0.23.0-git+03452b628
	2026-08-03T07:48:03.681489603Z [INFO] game/GameLoop.cc: Version #:     Build 04.12.02
	2026-08-03T07:48:03.692614473Z [ERROR] game/Utils/Cinematics.cc: Failed to open 'intro/splashscreen.smk': SGPFile::openInVfs: Vfs_open "intro/splashscreen.smk": entity not found
	2026-08-03T08:01:21.725141381Z [WARN] game/Strategic/Strategic_Movement.cc: Strategic Info Warning: Attempting to delete a persistant group.
	2026-08-03T08:01:25.849503022Z [ERROR] sgp/SGP.cc: Game has been terminated due to an unrecoverable error: Vehicle has no corresponding soldier
	Creating an emergency savegame succeeded (Error.sav).
	Please report this error with a description of the circumstances.  Do not forget to attach the savegame. (St13runtime_error)
	[1]    84179 IOT instruction  ja2
	ja2  107.90s user 13.63s system 15% cpu 13:23.33 total


Honeslty, I do not remember correctly if tanks were available in Meduna in vanila or if they were not availble as "vehicle". What is a bit strange is that a _destroyed_ tank is available as vehicle for the merc in the sector. The tanks are not "repair"-able, I have tried that out, also not sure here if they ever were to repair (tanks) in vanilla.

[Error.zip](https://github.com/user-attachments/files/30652510/Error.zip)

#### Steps to reproduce

1. Load Error save
2. In the "Map Screen", sector N6 assign a merc to a vehicle
3. In sector N6 there is also a (destroyed) tank in the vehicle list, choose that
4. Game crashes

Tried this out in other sectors in Meduna where destroyed tanks were availble. f.e.: N5 and N3, and it. In sector N3 there are 3 destroyed tanks available in the list as a vehicle. Optional savegame where all Meduna N sectors are available.

[Meduna-N-sectors.zip](https://github.com/user-attachments/files/30652504/Meduna-N-sectors.zip)

--- a/src/game/Tactical/Soldier_Create.cc
+++ b/src/game/Tactical/Soldier_Create.cc
@@ -1,7 +1,6 @@
 #include "Soldier_Create.h"
 #include "ItemModel.h"
 #include "Overhead.h"
-#include "Soldier_Macros.h"
 #include "Soldier_Profile.h"
 #include "Animation_Control.h"
 #include "Animation_Data.h"
@@ -47,6 +46,7 @@
 
 #include <algorithm>
 #include <iterator>
+#include <stdexcept>
 #include <math.h>
 
 // THESE 3 DIFFICULTY FACTORS MUST ALWAYS ADD UP TO 100% EXACTLY!!!
@@ -979,9 +979,14 @@ void InternalTacticalRemoveSoldier(SOLDIERTYPE& s, BOOLEAN const fRemoveVehicle)
 
 	if (s.ubScheduleID) DeleteSchedule(s.ubScheduleID);
 
-	if (EnterableVehicle(s) && fRemoveVehicle)
+	if ((s.uiStatusFlags & SOLDIER_VEHICLE) && fRemoveVehicle)
 	{
-		RemoveVehicleFromList(GetVehicle(s.bVehicleID));
+		try {
+			RemoveVehicleFromList(GetVehicle(s.bVehicleID));
+		}
+		catch (std::logic_error const&) {
+			// GetVehicle can fail for tanks added in map editor, issue #1933.
+		}
 	}
 
 	if (s.ubBodyType == CROW) HandleCrowLeave(&s);
@@ -2092,7 +2097,7 @@ void TrashAllSoldiers(int const team)
 }
 
 
-static UINT8 GetLocationModifier(UINT8 ubSoldierClass)
+static UINT8 GetLocationModifier(UINT8)
 {
 	UINT8 ubLocationModifier;
 	UINT8 ubPalaceDistance;

--- a/src/game/Tactical/Vehicles.cc
+++ b/src/game/Tactical/Vehicles.cc
@@ -142,6 +142,7 @@ bool IsThisVehicleAccessibleToSoldier(SOLDIERTYPE const& s, VEHICLETYPE const& v
 	return !s.fBetweenSectors &&
 		!v.fBetweenSectors &&
 		s.sSector == v.sSector &&
+		!v.fDestroyed &&
 		OKUseVehicle(GCM->getVehicle(v.ubVehicleType)->profile);
 }
 

